### ISSUE TITLE
Changefeed won't start via Web UI with `.limit(...)` in query (2.1.4, OS X)

### ISSUE THREAD

**Issue by dalanmiller** [rethinkdb/rethinkdb#4852]
This query will run:

```
`r.db("test").table("players").orderBy({index:r.desc("score")}).changes()`
```

However, this query will return the correct results but the changefeed won't start:

```
`r.db("test").table("players").orderBy({index:r.desc("score")}).limit(100).changes()`
```

[Here is a video example of this happening](https://www.dropbox.com/s/vtv6vt91agnz1gc/changefeed_not_starting.mov?dl=0)

**Comment by danielmewes** [comment]
That's weird. Could be related to https://github.com/rethinkdb/rethinkdb/issues/4850 as well.

**Comment by dalanmiller** [comment]
Originally, I did this on Chrome but this also occurs on Safari. I don't think it's a browser issue.

**Comment by coffeemug** [comment]
Is this reproducible in the drivers or is it purely a webui issue?

**Comment by dalanmiller** [comment]
@coffeemug - Ran same query via Python driver and it's working as expected.

Ran into another issue where I ran the query, and it just never seems to get a response from the server. I see that without running any query that it is doing some sort of heartbeat to the server for I assume information on the different nodes currently connected in the cluster. But after running the query, it pings one constantly as seen here (for you @deontologician):

![for_josh](https://cloud.githubusercontent.com/assets/863286/9946730/09c7490c-5d4a-11e5-9494-4791f3301b61.gif)

**Comment by deontologician** [comment]
Thank you :100:

**Comment by danielmewes** [comment]
I think we should re-test this once https://github.com/rethinkdb/rethinkdb/issues/4850 is fixed, because it might be a side effect of that bug.
If this still persists, we have to look into it more.

The HTTP interface that the web UI is using uses a 500ms timeout for each cursor CONTINUE request, which is what those "heartbeats" are coming from.

**Comment by dalanmiller** [comment]
Although it's working fine in the driver, I think this is pretty important considering we give this exact example in our docs and slides. 

https://rethinkdb.com/api/python/changes/

**Comment by coffeemug** [comment]
It's definitely very important to fix asap; we do things like demos of this all the time, it's a flagship feature of the product, and should definitely work.

BTW, can you reproduce it if the limit is smaller (e.g. 5 or 10 docs)?

**Comment by dalanmiller** [comment]
So interestingly if I run:

```
r.db("test").table("players").orderBy({index: "score"}).limit(40).changes()
```

It just halts and looks like this:

![screenshot 2015-09-17 15 13 43](https://cloud.githubusercontent.com/assets/863286/9947405/b7f81ffc-5d4e-11e5-9772-de3e690cd489.png)

However, if I reduce the limit by one to `39`, as in:

```
r.db("test").table("players").orderBy({index: "score"}).limit(39).changes()
```

Then it appears like the gif I posted above with the spinning icon.  [Here's a video for your entertainment and assistance](https://www.dropbox.com/s/8cz2ka0evoubfm4/limit_40.mov?dl=0) :open_mouth:!

**Comment by deontologician** [comment]
It looks like subsequent batches are causing the issue, this may be related to adding profile information to http queries

**Comment by deontologician** [comment]
Ok, this was really dumb. In order for the dataexplorer to detect we're dealing with a changefeed and use the right logic, it checks the cursor's `toString()` output. Specifically, it checks if it's `"[object Feed]"` or `"[object AtomFeed]"` and this happens to be a query that results in an `"[object OrderByLimitFeed]"`

**Comment by coffeemug** [comment]
We should totally add a predicate in the drivers on cursors to check for this stuff. Like `cursor.isFeed()` or something.

**Comment by danielmewes** [comment]
See�#4857 for adding explicit information on whether something is a feed or not.�

**Comment by coffeemug** [comment]
Ah, cool!

**Comment by deontologician** [comment]
Fixed in f59feae
